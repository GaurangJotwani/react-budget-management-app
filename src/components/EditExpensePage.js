import React from "react";
import ExpenseForm from "./ExpenseForm";
import { connect } from "react-redux";
import { editExpense, startRemoveExpense } from "../actions/expenses";

export class EditExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.editExpense(this.props.match.params.id, expense);
    this.props.history.push("/");
  };
  onClick = () => {
    this.props.startRemoveExpense({ id: this.props.match.params.id });
    this.props.history.push("/");
  };
  render() {
    return (
      <div>
        <h1>Edit Expense</h1>
        <ExpenseForm onSubmit={this.onSubmit} expense={this.props.expense} />
        <button onClick={this.onClick}>Remove</button>
      </div>
    );
  }
}

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find(
      (expense) => expense.id === props.match.params.id
    ),
  };
};

const mapDispatchToProps = (dispatch) => ({
  editExpense: (id, expense) => dispatch(editExpense(id, expense)),
  startRemoveExpense: (id) => dispatch(startRemoveExpense(id)),
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
