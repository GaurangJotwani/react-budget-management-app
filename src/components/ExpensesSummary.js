import React from "react";
import { connect } from "react-redux";
import selectExpenses from "../selectors/expenses";
import getExpensesTotal from "../selectors/expenses-total";
import numeral from "numeral";

export const ExpensesSummary = ({ count, total }) => {
  if (count === 0) {
    return null;
  }

  return (
    <h1>
      Viewing {count} {count === 1 ? "expense" : "expenses"} totalling{" "}
      {numeral(total).format("$0,0.00")}{" "}
    </h1>
  );
};

const mapStateToProps = (state) => {
  return {
    total:
      getExpensesTotal(selectExpenses(state.expenses, state.filters)) / 100,
    count: selectExpenses(state.expenses, state.filters).length,
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
