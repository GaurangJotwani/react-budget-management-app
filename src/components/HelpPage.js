import React from 'react'

const HelpPage = () => <div>This is from my help component page</div>;

export default HelpPage