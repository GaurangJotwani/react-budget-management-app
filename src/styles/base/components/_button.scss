.button {
  color: white;
  background: $blue;
  padding: $s-size;
  border: none;
  font-weight: 300;
  font-size: $font-size-large;
  display: inline-block;
  text-decoration: none;
  line-height: 1;
}

.button--link {
  background: none;
}

.button--secondary{
  background: #888;
}