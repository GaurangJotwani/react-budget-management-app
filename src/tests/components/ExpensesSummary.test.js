import React from "react";
import { shallow } from "enzyme";
import { ExpensesSummary } from "../../components/ExpensesSummary";
import toJson from "enzyme-to-json";

test("should render ExpensesSummary with empty message", () => {
  const wrapper = shallow(<ExpensesSummary count={0} total={0} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

test("should render ExpensesSummary with 1 expense", () => {
  const wrapper = shallow(<ExpensesSummary count={1} total={94.34} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

test("should render ExpensesSummary with expenses", () => {
  const wrapper = shallow(<ExpensesSummary count={2} total={94.34} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
