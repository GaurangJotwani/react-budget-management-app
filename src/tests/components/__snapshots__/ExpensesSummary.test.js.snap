// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with 1 expense 1`] = `
<h1>
  Viewing 
  1
   
  expense
   totalling
   
  $94.34
   
</h1>
`;

exports[`should render ExpensesSummary with empty message 1`] = `""`;

exports[`should render ExpensesSummary with expenses 1`] = `
<h1>
  Viewing 
  2
   
  expenses
   totalling
   
  $94.34
   
</h1>
`;
